traitlets/traitlets.py

```diff
@@ -783,13 +830,12 @@ def setup_class(cls, classdict):
 def observe(*names, **kwargs):
     """A decorator which can be used to observe Traits on a class.
 
-    The handler passed to the decorator will be called with one ``change``
-    dict argument. The change dictionary at least holds a 'type' key and a
-    'name' key, corresponding respectively to the type of notification and the
-    name of the attribute that triggered the notification.
+    The handler for the decorator must accept one ``change`` argument, into which is
+    passed a :class:`EventReport` instance that holds (at minumum) a 'type' key/attribute
+    corresponding to the type of event being reported. Other keys/attributes may be
+    included depending on the value of 'type'.
```

Comment:
This has removed the 'name' field, which is still guaranteed. It should also be clear that the object is still a considered a dictionary with attribute-access, rather than making any reference to EventReport, which is not a user-facing API.